custom_components/ha-google-home/sensor.py

```diff
@@ -88,6 +94,51 @@ def _get_alarms_data(self):
             format_alarm_information(alarm)
             for alarm in self.coordinator.data[self._name][LABEL_ALARMS]
         ]
+        alarms = sort_list_by_firetime(alarms)
+        return alarms
+
+
+class GoogleHomeNextAlarmSensor(GoogleHomeNextAlarmEntity):
+    """Representation of a Sensor."""
+
+    def __init__(self, coordinator, entry, device_name, alarms):
+        """Initialize the sensor."""
+        super().__init__(coordinator, entry)
+        self._name = device_name
+
+    @property
+    def state(self):
+        alarms = self._get_alarm_data()
+        state = alarms[0][LOCAL_TIME_ISO] if len(alarms) else STATE_OFF
+        # The first one will always be the closet one
+        # as we have sorted the list in _get_alarm_data()
+        return state
```

Comment:
Multiple changes here:
```suggestion
        # The first one will always be the closest one
        # as we have sorted the list in _get_alarm_data()
        return alarms[0][LOCAL_TIME_ISO] if alarms else STATE_OFF
```